server/sonar-web/src/main/js/apps/users/components/TokensForm.tsx

```diff
@@ -76,27 +71,27 @@ export default class TokensForm extends React.PureComponent<Props, State> {
     );
   };
 
-  handleCloseClick = (evt: React.SyntheticEvent<HTMLAnchorElement>) => {
-    evt.preventDefault();
-    this.handleClose();
-  };
-
-  handleClose = () => {
-    if (this.state.hasChanged) {
-      this.props.onUpdateUsers();
-    }
-    this.props.onClose();
-  };
-
   handleGenerateToken = (evt: React.SyntheticEvent<HTMLFormElement>) => {
     evt.preventDefault();
     if (this.state.newTokenName.length > 0) {
       this.setState({ generating: true });
-      generateToken({ name: this.state.newTokenName, login: this.props.user.login }).then(
+      generateToken({ name: this.state.newTokenName, login: this.props.login }).then(
         newToken => {
           if (this.mounted) {
-            this.fetchTokens();
-            this.setState({ generating: false, hasChanged: true, newToken, newTokenName: '' });
+            this.setState(
+              state => {
+                const tokens = [
```

Comment:
:+1: Nice!
I took the lazy path when I did it :grin:

Reply:
Actually it was impossible before (without `createdAt` field in the response of `api/user_tokens/generate`).

Reply:
Ah yes that's right! :relieved: